src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteSpec.groovy

```diff
@@ -80,7 +82,12 @@ class AutoRerouteSpec extends HealthCheckSpecification {
         def portDown = antiflap.portDown(isl.dstSwitch.dpId, isl.dstPort)
 
         then: "The flow becomes 'Down'"
-        Wrappers.wait(rerouteDelay + WAIT_OFFSET) { assert northbound.getFlowStatus(flow.id).status == FlowState.DOWN }
+        Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
+            assert northbound.getFlowStatus(flow.id).status == FlowState.DOWN
+            assert northbound.getFlowHistory(flow.id).find {
+                it.action == REROUTE_ACTION && it.taskId =~ (/.+ : retry #1 ignore_bw true/)
+            }?.payload.last().action == REROUTE_FAIL
```

Comment:
should be either `}?.payload?.last()?.action` or just don't use safe navigation at all

Reply:
I thought one safe navigation would be enough.
fixed.